source/server/overload_manager_impl.cc

```diff
@@ -75,22 +75,29 @@ class ScaledTriggerImpl final : public OverloadAction::Trigger {
  */
 class ThreadLocalOverloadStateImpl : public ThreadLocalOverloadState {
 public:
+  ThreadLocalOverloadStateImpl(const NamedOverloadActionSymbolTable& action_symbol_table)
+      : action_symbol_table_(action_symbol_table),
+        actions_(action_symbol_table.size(), OverloadActionState(0)) {}
+
   const OverloadActionState& getState(const std::string& action) override {
-    auto it = actions_.find(action);
-    if (it == actions_.end()) {
-      it = actions_.insert(std::make_pair(action, OverloadActionState::inactive())).first;
+    if (auto symbol = action_symbol_table_.find(action); symbol != absl::nullopt) {
+      return actions_[symbol->index()];
     }
-    return it->second;
+    return always_inactive_;
   }
 
-  void setState(const std::string& action, OverloadActionState state) {
-    actions_.insert_or_assign(action, state);
+  void setState(NamedOverloadActionSymbolTable::Symbol action, OverloadActionState state) {
+    actions_[action.index()] = state;
   }
 
 private:
-  absl::node_hash_map<std::string, OverloadActionState> actions_;
+  static const OverloadActionState always_inactive_;
```

Comment:
Does this initialize to `OverloadActionState(0)` by default, and that implies "always_inactive"? Maybe it would be worth explicitly instantiating this in the ctor with a comment.